service/matching/taskListManager.go

```diff
@@ -323,6 +407,13 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 		}()
 	}
 
+	// Wait till long poll expiration for token. If token acquired, proceed.
+	if err := c.rateLimiter.Wait(c.config.LongPollExpirationInterval); err != nil {
+		c.metricsClient.IncCounter(scope, metrics.ThrottleErrorCounter)
+		msg := fmt.Sprintf("TaskList dispatch exceeded limit: %s", err.Error())
+		// TODO: It's the client that's busy, not the server. Doesn't fit into any other error.
+		return nil, createServiceBusyError(msg)
```

Comment:
I don't think we want to return service busy error. Also it is going to be returned even if there is no backlog as we throttle before trying to match. From the poller point of view it is ErrNoTasks.

Reply:
Same as above, want to differentiate between truly no tasks vs tasks being throttled. Either way I need a new thrift type so user can react to it. ErrNewTasks is not a type, it just has a specific string message.